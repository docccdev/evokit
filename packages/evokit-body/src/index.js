import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
