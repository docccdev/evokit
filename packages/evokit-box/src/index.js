// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
