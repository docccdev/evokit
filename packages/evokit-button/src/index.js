import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
