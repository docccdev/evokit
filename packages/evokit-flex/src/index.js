import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
