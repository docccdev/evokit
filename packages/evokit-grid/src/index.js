import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
