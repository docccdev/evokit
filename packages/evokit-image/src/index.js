import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
