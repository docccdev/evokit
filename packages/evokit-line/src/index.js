import { createBlock } from 'evokit';

export const Line = createBlock('hr', 'line', [
    'display',
    'indent',
    'style',
    'color',
]);
