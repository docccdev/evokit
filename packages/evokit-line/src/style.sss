@import './mixins.sss'

@custom-media --ek-line-media-small only screen and (min-width: 480px)
@custom-media --ek-line-media-medium only screen and (min-width: 768px)
@custom-media --ek-line-media-large only screen and (min-width: 960px)
@custom-media --ek-line-media-wide only screen and (min-width: 1200px)
@custom-media --ek-line-media-huge only screen and (min-width: 1400px)

:root
    // prop 'line-indent'
    --ek-line-indent-xxs: 5px
    --ek-line-indent-xs: 10px
    --ek-line-indent-s: 15px
    --ek-line-indent-m: 20px
    --ek-line-indent-l: 25px
    --ek-line-indent-xl: 30px
    --ek-line-indent-xxl: 35px

.line
    display: block
    border-width: 1px 0 0
    border-style: solid
    border-color: transparent
    margin: 0

/**
 * Style:
 *
 *  - line_display_*
 *
 **/

@each $value in (block, none)
    @mixin media .line_display_$(value)
        display: $value

/**
 * Indent:
 *
 *  - line_indent_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .line_indent_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0
        @else
            margin-top: var(--ek-line-indent-$(value))
            margin-bottom: var(--ek-line-indent-$(value))

/**
 * Style:
 *
 *  - line_style_*
 *
 **/

@each $value in (solid, dotted, dashed)
    @mixin media .line_style_$(value)
        border-top-style: $value
