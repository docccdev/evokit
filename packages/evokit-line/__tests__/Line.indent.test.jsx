import React from 'react';
import { shallow } from 'enzyme';
import { Line } from '../src';

describe('<Line line-indent="..." />', () => {
    it('props line-indent', () => {
        const wrapper = shallow(<Line line-indent='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<hr class="ek-line ek-line_indent_value"/>');
    });
});
