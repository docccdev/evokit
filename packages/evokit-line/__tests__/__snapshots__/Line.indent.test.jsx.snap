// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Line line-indent="..." /> props line-indent 1`] = `
<hr
  className="ek-line ek-line_indent_value"
/>
`;
